code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/lb/LoadBalancerHostValidationFilter.java

```diff
@@ -0,0 +1,71 @@
+package io.cattle.platform.iaas.api.filter.lb;
+
+import io.cattle.platform.core.constants.LoadBalancerConstants;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.core.model.LoadBalancer;
+import io.cattle.platform.core.model.LoadBalancerHostMap;
+import io.cattle.platform.iaas.api.filter.common.AbstractDefaultResourceManagerFilter;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.validation.ValidationErrorCodes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+public class LoadBalancerHostValidationFilter extends AbstractDefaultResourceManagerFilter {
+    
+    GenericMapDao mapDao;
+
+    private static final Map<String, Boolean> actions;
+    static
+    {
+        actions = new HashMap<>();
+        actions.put(LoadBalancerConstants.ACTIONT_ADD_HOST.toLowerCase(), true);
+        actions.put(LoadBalancerConstants.ACTIONT_REMOVE_HOST.toLowerCase(), false);
+    }
+    
+    @Override
+    public String[] getTypes() {
+        return new String[] { LoadBalancerConstants.OBJECT_LB };
```

Comment:
You don't actually have to do both getTypes() and getTypeClasses().  You can just return new String[0] here because the getTypeClasses() below will match the LB type.  The string variation is only really helpful for subtypes.